src/DynamoCoreWpf/ViewModels/Core/DynamoViewModelEventArgs.cs

```diff
@@ -103,4 +103,18 @@ public ImageSaveEventArgs(string path)
         }
     }
 
+    public class IconRequestEventArgs : EventArgs
+    {
+        public string IconAssembly { get; set; }
+
+        public string IconFullPath { get; set; }
+
+        public BitmapSource Icon { get; set; }
```

Comment:
Make all of the above `private set`, and then add a new method to set `BitmapSource` called `internal void SetIcon`.